feat: only allow selected users

artemv opened the pull request:
We don't want to allow rolling out features to everyone. Also, secure the features list too (for https://github.com/fiverr/rollout_dashboard/pull/17).

artemv committed:
feat: only allow selected users

We don't want to allow rolling out features to everyone. Also, secure the features list too.

yossi-eynav commented on rollout_service.rb:
```diff
@@ -11,13 +11,18 @@ def authenticate!
           raise 'Unauthorized user, this domain is not allowed'
         end
 
+        allowed = (ENV['ALLOWED_EMAILS'] || '').split(',')
```
Instead of accessing the ENV directly, set this list as a global.
See `$google_oauth_allowed_domain ` as a reference

yossi-eynav commented on rollout_service.rb:
```diff
@@ -11,13 +11,18 @@ def authenticate!
                                  body: {id_token: params[:id_token]})
 
-        raise 'Bad response from server' if response.code != 200
+        raise "Bad response from server: #{response.inspect}" if response.code != 200
```
👍

yossi-eynav commented on rollout_service.rb:
```diff
@@ -11,13 +11,18 @@ def authenticate!
 
+        allowed = (ENV['ALLOWED_EMAILS'] || '').split(',')
+        if !allowed.empty? && !allowed.include?(response_body['email'])
+          puts "#{response_body['email'].inspect} is not allowed (#{allowed.inspect})"
```
what's the benefit?

artemv replied:
Currently rollout_service/rollout_dashboard is allowing anyone with a Gmail account to make changes - which is not what one would expect from an admin tool. This change fixes it, it makes it check the user account against allowed emails list set in ALLOWED_EMAILS env var.

yossi-eynav replied:
@artemv 
I meant what's the benefit with line 23. (the log  output line)

yossi-eynav commented on README.md:
````diff
@@ -24,4 +24,8 @@ This service works great with [Rollout-Dashboard](https://github.com/fiverr/roll
+```
+ALLOWED_EMAILS="user1@gmail.com,user2@gmail.com" bundle exec rackup -p 4000
+```
+Specify proper ALLOWED_EMAILS and service port. In this example the service will listen on port 4000.
````
1- please add a section in the FAQ that you can restrict access to specific users
2- maybe we should authenticate all calls to the API? 
3- since you're adding it to a READ operation, edit the README file

artemv replied:
2 - in scope of this PR I only want to make sure the main screen of Dashboard (i.e. the feature list) is secured.

artemv committed:
refactor: moved ALLOWED_USERS_EMAILS to global

.. as Yossi asked in https://github.com/fiverr/rollout_service/pull/9

also more docs for ALLOWED_USERS_EMAILS

artemv committed:
docs: corrected re ALLOWED_USERS_EMAILS
